### PR TITLE
Add standardized model card for facebook/data2vec-audio-base-960h

### PR DESCRIPTION
**Pull request by boy397**
This PR adds a standardized model card for [`facebook/data2vec-audio-base-960h`](https://huggingface.co/facebook/data2vec-audio-base-960h) as part of issue [#36979](https://github.com/huggingface/transformers/issues/36979).

### What's included:
- Friendly, accessible model description
- Pipeline usage example for speech recognition
- AutoModel + AutoProcessor example
- Info about quantization (if applicable)
- Uses the latest formatting standards for model cards

This model card will improve discoverability and usability for the speech model by aligning with Hugging Face documentation standards.

### CODE REVIEW

**Comment by boy397** [comment]
Hi @stevhliu I have made the changes to the data2vec file in the docs directory as you have told. Please tell me if there is anything else i have to do..
Thank you

**Review by stevhliu — COMMENTED** [review]
Thanks for getting started! I recommend reviewing the original post in https://github.com/huggingface/transformers/issues/36979 to see how you should format the model card and what type of information to include. It is missing quite a bit at the moment

**Comment by boy397** [comment]
Thank you for the feedback . 
I will  work on the format of the model card and the information

**Comment by boy397** [comment]
Hi @stevhliu I was trying to fix the model structure that you have mentioned and made the changes.
But after I pushed my changes it is showing here that (check_repository_consistency and setup_and_quality) failed.
Due to which I tries to run this in my local so that the problem can be solves for which,
1. .my-env\Scripts\activate
2. python utils/check_repo_consistency.py --fix_and_commit but it was not running
I can't understand what is the issue that I am facing. Please provide some way to solve the above problem.

Thank You

**Comment by boy397** [comment]
Hi @stevhliu can you please help me with the problem.

File: docs/source/en/model_doc/data2vec.md [hunk 1]
```diff
@@ -14,221 +14,83 @@ rendered properly in your Markdown viewer.
 
 -->
 
-# Data2Vec
-
-<div class="flex flex-wrap space-x-1">
-<img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-DE3412?style=flat&logo=pytorch&logoColor=white">
-<img alt="FlashAttention" src="https://img.shields.io/badge/%E2%9A%A1%EF%B8%8E%20FlashAttention-eae0c8?style=flat">
-<img alt="SDPA" src="https://img.shields.io/badge/SDPA-DE3412?style=flat&logo=pytorch&logoColor=white">
+<div style="float: right;">
```
Comment (stevhliu) [inline docs/source/en/model_doc/data2vec.md:17]: Missing the FlashAttention and SDPA badges

File: docs/source/en/model_doc/data2vec.md [hunk 2]
```diff
@@ -14,221 +14,83 @@ rendered properly in your Markdown viewer.
...
-
-This model was contributed by [edugp](https://huggingface.co/edugp) and [patrickvonplaten](https://huggingface.co/patrickvonplaten).
-[sayakpaul](https://github.com/sayakpaul) and [Rocketknight1](https://github.com/Rocketknight1) contributed Data2Vec for vision in TensorFlow.
-
-The original code (for NLP and Speech) can be found [here](https://github.com/pytorch/fairseq/tree/main/examples/data2vec).
-The original code for vision can be found [here](https://github.com/facebookresearch/data2vec_vision/tree/main/beit).
+# Data2Vec
 
-## Usage tips
+[Data2Vec](https://huggingface.co/papers/2202.03555) is a self-supervised model developed by Meta AI that works across multiple modalities including speech, vision, and text. This model card focuses on the audio variant: `data2vec-audio-base-960h`, which learns contextualized representations directly from raw audio using a masked prediction task.
```
Comment (stevhliu) [inline docs/source/en/model_doc/data2vec.md]:
```suggestion
[Data2Vec](https://huggingface.co/papers/2202.03555) is a unified self-supervised learning framework that works across multiple modalities. It predicts contextualized latent representations that capture information from the entire input rather than modality-specific objectives like words or visual tokens.
```

File: docs/source/en/model_doc/data2vec.md [hunk 3]
```diff
@@ -14,221 +14,83 @@ rendered properly in your Markdown viewer.
...
 
-- Data2VecAudio, Data2VecText, and Data2VecVision have all been trained using the same self-supervised learning method.
-- For Data2VecAudio, preprocessing is identical to [`Wav2Vec2Model`], including feature extraction
-- For Data2VecText, preprocessing is identical to [`RobertaModel`], including tokenization.
-- For Data2VecVision, preprocessing is identical to [`BeitModel`], including feature extraction.
-- The `head_mask` argument is ignored when using all attention implementation other than "eager". If you have a `head_mask` and want it to have effect, load the model with `XXXModel.from_pretrained(model_id, attn_implementation="eager")`  
+You can find all the original Data2Vec checkpoints under the [Data2Vec](https://huggingface.co/facebook/data2vec-audio-base-960h) collection.
```
Comment (stevhliu) [inline docs/source/en/model_doc/data2vec.md]:
```suggestion
You can find all the original Data2Vec checkpoints under the [AI at Meta](https://huggingface.co/facebook/models?search=data2vec) organization.
```

File: docs/source/en/model_doc/data2vec.md [hunk 4]
```diff
@@ -14,221 +14,83 @@ rendered properly in your Markdown viewer.
...
 
-### Using Scaled Dot Product Attention (SDPA)
+> [!TIP]
+> This model was contributed by [facebook](https://huggingface.co/facebook).
+>
+> Click on the Data2Vec models in the right sidebar for more examples of how to apply Data2Vec to speech tasks.
```
Comment (stevhliu) [inline docs/source/en/model_doc/data2vec.md]:
```suggestion
> This model was contributed by [edugp](https://huggingface.co/edugp) and [patrickvonplaten](https://huggingface.co/patrickvonplaten). The TensorFlow version was contributed by [sayakpaul](https://github.com/sayakpaul) and [Rocketknight1](https://github.com/Rocketknight1).
>
> Click on the Data2Vec models in the right sidebar for more examples of how to apply Data2Vec to other tasks.
```

File: docs/source/en/model_doc/data2vec.md [hunk 5]
```diff
@@ -14,221 +14,83 @@ rendered properly in your Markdown viewer.
...
+The example below demonstrates how to transcribe audio using [`Pipeline`] or the [`AutoModel`] class.
 
-SDPA is used by default for `torch>=2.1.1` when an implementation is available, but you may also set 
-`attn_implementation="sdpa"` in `from_pretrained()` to explicitly request SDPA to be used.
+<hfoptions id="usage">
+<hfoption id="Pipeline">
 
-The SDPA implementation is currently available for the Data2VecAudio and Data2VecVision models.
+```python
+from transformers import pipeline
```
Comment (stevhliu) [inline docs/source/en/model_doc/data2vec.md:41]:
```py
import torch
from transformers import pipeline

pipeline = pipeline(model="facebook/data2vec-audio-base-960h", task="automatic-speech-recognition", torch_dtype=torch.float16, device=0)
pipeline("https://huggingface.co/datasets/Narsil/asr_dummy/resolve/main/mlk.flac")
```

File: docs/source/en/model_doc/data2vec.md [hunk 6]
```diff
@@ -14,221 +14,83 @@ rendered properly in your Markdown viewer.
...
-
-#### Training
-
-| num_training_steps | batch_size | image_size   | is_cuda | Time per batch (eager - s) | Time per batch (sdpa - s) | Speedup (%) | Eager peak mem (MB) | SDPA peak mem (MB) | Mem saving (%) |
-|--------------------|------------|--------------|---------|----------------------------|---------------------------|-------------|----------------------|--------------------|----------------|
-| 50                 | 2          | (1048, 640)  | True    | 0.996                      | 0.754                     | 32.147      | 6722.198            | 4264.653          | 57.626         |
 
-#### Inference
+</hfoption>
+<hfoption id="AutoModel">
```
Comment (stevhliu) [inline docs/source/en/model_doc/data2vec.md:47]:
```py
import torch
from datasets import load_dataset
from transformers import AutoProcessor, AutoModelForCTC

dataset = load_dataset("hf-internal-testing/librispeech_asr_demo", "clean", split="validation")
dataset = dataset.sort("id")
sampling_rate = dataset.features["audio"].sampling_rate

processor = AutoProcessor.from_pretrained("facebook/data2vec-audio-base-960h")
model = AutoModelForCTC.from_pretrained("facebook/data2vec-audio-base-960h", torch_dtype=torch.float16, attn_implementation="sdpa")

inputs = processor(dataset[0]["audio"]["array"], sampling_rate=sampling_rate, return_tensors="pt")
with torch.no_grad():
    logits = model(**inputs).logits
predicted_ids = torch.argmax(logits, dim=-1)

transcription = processor.batch_decode(predicted_ids)
transcription[0]
```

File: docs/source/en/model_doc/data2vec.md [hunk 7]
```diff
@@ -14,221 +14,83 @@ rendered properly in your Markdown viewer.
...
-- [Masked language modeling task guide](../tasks/masked_language_modeling)
-- [Multiple choice task guide](../tasks/multiple_choice)
+predicted_class_id = torch.argmax(logits).item()
+```
 
-**Data2VecAudio documentation resources**
-- [Audio classification task guide](../tasks/audio_classification)
-- [Automatic speech recognition task guide](../tasks/asr)
+</hfoption>
+<hfoption id="transformers CLI">
```
Comment (stevhliu) [inline docs/source/en/model_doc/data2vec.md:67]: You can remove the `transformers cli` section

File: docs/source/en/model_doc/data2vec.md [hunk 8]
```diff
@@ -14,221 +14,83 @@ rendered properly in your Markdown viewer.
...
-**Data2VecVision documentation resources**
-- [Image classification](../tasks/image_classification)
-- [Semantic segmentation](../tasks/semantic_segmentation)
+<!-- CLI not supported for this model -->
 
-If you're interested in submitting a resource to be included here, please feel free to open a Pull Request and we'll review it! The resource should ideally demonstrate something new instead of duplicating an existing resource.
+</hfoption>
+</hfoptions>
 
-## Data2VecTextConfig
```
Comment (stevhliu) [inline docs/source/en/model_doc/data2vec.md:124]: You don't need to change the API docs here, they should all remain

File: docs/source/en/model_doc/data2vec.md [hunk 9]
```diff
@@ -14,221 +14,83 @@ rendered properly in your Markdown viewer.
...
+## Notes
 
-[[autodoc]] Data2VecTextConfig
+- This model supports audio feature extraction and is pretrained on LibriSpeech.
+- Input audio must be sampled at 16kHz.
+- It's recommended to use the corresponding AutoProcessor to ensure correct preprocessing.
```
Comment (stevhliu) [inline docs/source/en/model_doc/data2vec.md]:
```suggestion
- [`Data2VecAudioModel`] preprocessing is identical to [`Wav2Vec2Model`], [`Data2VecText`] preprocessing and tokenization is identical to [`RobertaModel`], and [`Data2VecVision`] preprocessing and feature extraction is identical to [`BeitModel`].
- The `head_mask` argument is ignored if the attention implementation isn't `"eager"`. To use `head_mask`, make sure `attn_implementation="eager"`.
```

File: docs/source/en/model_doc/data2vec.md [hunk 10]
```diff
@@ -14,221 +14,83 @@ rendered properly in your Markdown viewer.
...
-|                  1 |            0.011 | ±0.3%         |         3.76143e+08 |           0.01  | ±0.3%        |        3.74397e+08 |          1.101 |               0.466 |
-|                  4 |            0.014 | ±0.1%         |         4.02756e+08 |           0.012 | ±0.2%        |        3.91373e+08 |          1.219 |               2.909 |
-|                 16 |            0.046 | ±0.3%         |         4.96482e+08 |           0.035 | ±0.2%        |        4.51017e+08 |          1.314 |              10.081 |
-|                 32 |            0.088 | ±0.1%         |         6.23903e+08 |           0.067 | ±0.1%        |        5.32974e+08 |          1.33  |              17.061 |
+```python
+from transformers import AutoProcessor, AutoModelForAudioClassification
+import torch
+import torchaudio
 
-## Resources
```
Comment (stevhliu) [inline docs/source/en/model_doc/data2vec.md:98]: Add a `## Resources` section after `## Notes`.

```md
## Resources
- Refer to this [notebook](https://colab.research.google.com/github/sayakpaul/TF-2.0-Hacks/blob/master/data2vec_vision_image_classification.ipynb) for an example of how to fine-tune [`TFData2VecVisionForImageClassification`] on a custom dataset.
```

**Review by stevhliu — COMMENTED** [review]
Thanks! You can remove the `README.md` file here

**Comment by boy397** [comment]
@stevhliu thanks for the help

**Comment by boy397** [comment]
Hi @stevhliu I am confused that how to work on this after you have made the changes to this code. I have never faced this situation, and for that i have been stucked in this for a long time but couldn't figure it out.
Can you please suggest me what should I do after you have made changes and how should i make the commit after it . Please help needed.

**Comment by stevhliu** [comment]
Hey, it looks like there are still some unresolved changes! The failed checks are a result of removing the API sections.

- On your branch, you should resolve the conflict by rebasing on main.
- Apply the remaining unresolved suggestions from above (like this [one](https://github.com/huggingface/transformers/pull/39368#discussion_r2223250629)). 
- Force push (`git push -f`) your changes here

**Comment by boy397** [comment]
Thank you for your help, now i am working on it.

**Comment by boy397** [comment]
@stevhliu after i made the forced push here ci is getting failed. can you please tell me the reason of it?

**Comment by boy397** [comment]
This is the code that i have provided but it is not getting reflected..


<div style="float: right;">
    <div class="flex flex-wrap space-x-1">
        PyTorch
    </div>
</div>

# Data2Vec

[Data2Vec](https://huggingface.co/papers/2202.03555) is a unified self-supervised learning framework that works across multiple modalities. It predicts contextualized latent representations that capture information from the entire input rather than modality-specific objectives like words or visual tokens.

You can find all the original Data2Vec checkpoints under the [AI at Meta](https://huggingface.co/facebook/models?search=data2vec) organization.

> [!TIP]
> This model was contributed by [edugp](https://huggingface.co/edugp) and [patrickvonplaten](https://huggingface.co/patrickvonplaten). The TensorFlow version was contributed by [sayakpaul](https://github.com/sayakpaul) and [Rocketknight1](https://github.com/Rocketknight1).
>
> Click on the Data2Vec models in the right sidebar for more examples of how to apply Data2Vec to other tasks.

The example below demonstrates how to transcribe audio using [`Pipeline`] or the [`AutoModel`] class.

<hfoptions id="usage">
<hfoption id="Pipeline">

```
python
from transformers import pipeline
pipe = pipeline(model="facebook/data2vec-audio-base-960h", task="automatic-speech-recognition")
pipe("path/to/audio.wav")
```

</hfoption>
<hfoption id="AutoModel">

```
python
from transformers import AutoProcessor, AutoModelForAudioClassification
import torch
import torchaudio
processor = AutoProcessor.from_pretrained("facebook/data2vec-audio-base-960h")
model = AutoModelForAudioClassification.from_pretrained("facebook/data2vec-audio-base-960h")

waveform, sample_rate = torchaudio.load("path/to/audio.wav")
inputs = processor(waveform, sampling_rate=sample_rate, return_tensors="pt")

with torch.no_grad():
    logits = model(**inputs).logits

predicted_class_id = torch.argmax(logits).item()
```

</hfoption>
<hfoption id="transformers CLI">



</hfoption>
</hfoptions>

## Notes

- [`Data2VecAudioModel`] preprocessing is identical to [`Wav2Vec2Model`], [`Data2VecText`] preprocessing and tokenization is identical to [`RobertaModel`], and [`Data2VecVision`] preprocessing and feature extraction is identical to [`BeitModel`].
- The `head_mask` argument is ignored if the attention implementation isn't `"eager"`. To use `head_mask`, make sure `attn_implementation="eager"`.

## Resources
- Refer to this [notebook](https://colab.research.google.com/github/sayakpaul/TF-2.0-Hacks/blob/master/data2vec_vision_image_classification.ipynb) for an example of how to fine-tune [`TFData2VecVisionForImageClassification`] on a custom dataset.

## Data2VecAudioConfig

[[autodoc]] Data2VecAudioConfig
- all

## Data2VecAudioModel

[[autodoc]] Data2VecAudioModel
- forward

## Data2VecAudioPreTrainedModel

[[autodoc]] Data2VecAudioPreTrainedModel

## Data2Vec specific outputs

[[autodoc]] models.data2vec.modeling_data2vec_audio.Data2VecAudioBaseModelOutput




Anything else do I have to change in this code? I can't find the error please help me figure this out..

**Comment by stevhliu** [comment]
Thank you so much for the time and effort you put into this—I really appreciate it. With the number of models in Transformers growing so quickly, I’ve been looking into ways to automate model card generation to keep up with the pace. For now, I’ll be closing this PR, but I’m very grateful for your interest in this project!

Thanks again 🤗

**Comment by boy397** [comment]
Thank you for the clarification and kind words! 🙏 I just wanted to check if there was anything I could have done better in this PR, or if it was closed purely because of the upcoming automation. I’d also love to know how I can improve myself to contribute more effectively in the future. Please let me know if I can help with the automation direction as well. 🤗

🤗 Any suggestions for areas where I can make contributions next would be super helpful!
